inst/pages/multimedia.qmd

```diff
@@ -0,0 +1,589 @@
+mediators, followed by pathways abundances.
+
+Generally, we will proceed through the following key steps:
+
+1.  Prepare the microbiome taxonomic data from the iHMP IBDMDB project.
+
+2.  Calculate dysbiosis scores as the outcome.
+
+3.  Define the mediation analysis data structure.
+
+4.  Fit the multimodal mediation model using the R package multimedia.
+
+5.  Interpret both the overall indirect effects and the mediator-specific 
+indirect effects.
+
+6.  Visualize the results using forest plots, histograms, and rankings to 
+prioritize findings.
+
+7.  Repeat the process for the iHMP microbial pathways.
+
+### Performing multimodal mediation analysis for iHMP species relative abundance {#sec-relative-abundance}
+
+We begin by loading the relative abundance data and calculating dysbiosis scores 
+for each sample based on Bray–Curtis dissimilarity from a healthy reference set. 
+This represents deviation from a healthy microbiome, where higher dysbiosis 
+scores indicate greater microbial imbalance. We then subset to IBD subjects with 
+samples collected at both baseline (visit 1) and post-treatment (visit 21).
+
+```{r}
+#| label: iHMP_data_preprocessing
+#| message: false
+
+##################
+# Load libraries #
+##################
+
+library(curatedMetagenomicData)
+library(SummarizedExperiment)
+library(dplyr)
+library(vegan)
+library(tidyverse)
+library(multimedia)
+library(stringr)
+library(ggplot2)
+library(mia)
+
+##################
+# Load iHMP data #
+##################
+
+# Load data
+tse <- curatedMetagenomicData(
+  "HMP_2019_ibdmdb.relative_abundance", 
+  rownames = "short",
+  dryrun = FALSE
+  )[[1]]
+
+# Assign SampleID for matching
+tse[["SampleID"]] <- colnames(tse)
+
+# Convert relative_abundance assay to relabundance (which is in [0,1] interval)
+tse <- transformAssay(tse, assay.type="relative_abundance", method="relabundance")
+
+########################
+# Reference nonIBD set #
+########################
+
+# Reference set: healthy
+tse$disease_binary <- tse$disease == "healthy"
+
+########################################
+# Calculate Bray-Curtis dissimilarity #
+########################################
+
+# Bray-Curtis dissimilarity
+diss <- as.matrix(getDissimilarity(tse, method = "bray", na.rm=TRUE, assay.type = "relabundance"))
+
+#################################
+# Calculate the dysbiosis score #
+#################################
+
+# Calculate dysbiosis score for each sample
+# For each sample i, we compute the median distance between sample i and all reference samples in `ref_set`
+sample_ids <- tse[["SampleID"]]
+
+tse[["dysbiosis"]] <- sapply(seq_along(tse$disease_binary), function(i) {
+
+  # Logical vector indicating all other reference samples (excluding i)
+  ref_others <- tse$disease_binary & (sample_ids != sample_ids[i])
+  
+  # Compute median distance between sample i and these reference samples
+  median(diss[i, ref_others], na.rm = TRUE)
+
+})
```

Comment:
Can you check if the functions from the [dysbiosis package](https://microsud.github.io/dysbiosisR/articles/Introduction.html) could be applicable here?

Reply:
We do not want to convert to phyloseq. We could probably add SE support for this package relatively easily but maybe it is better not wait for that. Unless @microsud wants to comment on this?